articles/setting-up-cicd-for-python-packages-using-github-actions/index.md

```diff
@@ -0,0 +1,279 @@
+
+Copy and paste the following code snippet into `logic.py`.
+
+```python
+from datetime import datetime
+import pytz
+
+def area(location):
+    location = format_location(location)
+    for areas in pytz.all_timezones:
+        if location.lower() in areas.lower():
+            location = areas
+            tz = pytz.timezone(location)
+            date_now = datetime.now(tz)
+            formatted_date = date_now.strftime("%B %d, %Y %H:%M:%S")
+            print(f"{location} time: ", formatted_date)
+            break
+
+    else:
+        print("This location isn't on the tz database on Wikipedia")
+
+def area_zone(zone):
+    try:
+        zone = timezones(zone)
+        tz = pytz.timezone(zone)
+        date_now = datetime.now(tz)
+        formatted_date = date_now.strftime("%B %d, %Y %H:%M:%S")
+        print(f"{zone} time: ", formatted_date)
+
+    except Exception:
+        print("Timezone is not on the list. Consider using location instead.")
+
+def timezones(zone):
+	"""This function is used to handle situations of Daylight Saving Time that the standard library can't recognize."""
+    zones = {
+        "PDT": "PST8PDT",
+        "MDT": "MST7MDT",
+        "EDT": "EST5EDT",
+        "CDT": "CST6CDT",
+        "WAT": "Etc/GMT+1",
+        "ACT": "Australia/ACT",
+        "AST": "Atlantic/Bermuda",
+        "CAT": "Africa/Johannesburg",
+    }
+
+    try:
+        zones[zone]
+			
+    except:
+        return zone
+    return zones[zone]
+
+def format_location(location):
+    location = location.replace(" ", "_")
+    return location
+```
+
+Next we copy the following code snippet into `main.py`.
+
+```python
+import click
+from src.logic import area, area_zone
+
+@click.command()
+@click.option(
+    "--location",
+    help="This specifies the location you want to know the time. For example, Lagos or London",
+)
+@click.option(
+    "--zone",
+    help="The timezone information you need. Ensure it is properly capitalized, for example CET or WAT",
+)
+def main(location, zone):
+    if location:
+        area(location)
+    if zone:
+        area_zone(zone)
+
+if __name__ == "__main__":
+    main()
+```
+
+We leave the `__init__.py` empty.
+
+**Important things to note:**
+
+1. The `logic.py` uses the Python package *pytz* to understand the different time zones as it comes with the time zones in-built. We wrap the various functions in this file around *pytz* and its in-built functions. We also format some of the results to fit our end goal, a CLI for time zones.
+2. The `main.py` is where the magic happens as we build and design our CLI. The *click* library is used to build CLIs in Python (similar to *typer*, *fire*, and the in-built *argparse*). This library has functions that wrap around our previously created functions in `logic.py` to interface directly with a user on the command line.
+3. The `__init__.py` enables the `src` folder to be seen as a module as we imported some functions from *logic.py* into *main.py*.
```

Comment:
```suggestion
3. The `__init__.py` enables the `src` folder to be seen as a module as we imported some functions from `logic.py` into `main.py`.
```
Changing *logic.py* -> `logic.py`.

*logic.py* is being rendered as a hyperlink while markdown formatting.